mysql-test/t/percona_innodb_fake_changes.test

```diff
@@ -448,6 +445,56 @@ drop table t2;
 
 #-------------------------------------------------------------------------------
 #
+# Scenario is meant to check following scenarios:
+# SELECT is allowed to function normally if fake_changes = ON as it is readonly
+# CREATE...SELECT * from <table>; is allowed to work if CREATE is done using
+# other SE but SELECT may be working on InnoDB.
+#
+create table t1 (i int) engine=innodb;
+insert into t1 values (1), (2);
+set session innodb_fake_changes = 1;
+#
+select * from t1;
+set @@global.default_storage_engine=default;
```

Comment:
But you don't open any new connections to the server so your scenario is immune to global var change?